### PR TITLE
feat(cdk-v2): Adding build scripts for CDK v2

### PR DESCRIPTION
**Pull request by hnishar**
*Issue #, if available:* [348](https://github.com/awslabs/aws-solutions-constructs/issues/348)

*Description of changes:*

Adding build scripts for CDK v2 build

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: cbe0d746bc2faf03107f1a207204c0a37388e69a
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e281acbc-9c2f-437c-887a-047f38e25156%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 1ef438cf230570aba0b4733c6800f808c462c41f
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2f60ea5b-964c-4b14-9044-02c3fe39c8d6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 422bf208e924ce5d841a6f55c6bb7226d5c8adbe
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=44ea5332-bc2f-4c87-87b7-3ae39509f979%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: e1320d70e245f789ebbfcc9782053aec69644494
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5ab21b85-e302-41da-a6b7-2dbeae4ab76c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/tools/aws-cdk-migration/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (biffgaut) [inline source/tools/aws-cdk-migration/LICENSE:1]: Are we duplicating the cdk migration tool here?

Why would we do this?

File: deployment/v2/align-version.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env node
+// It will make following updates to package.json
+// 1 - align the version in a package.json file to the version of the repo
+// 2 - Remove all entries starting with @aws-cdk/* and constructs from "dependencies": { ... }
+// 3 - Remove all entries starting with @aws-cdk/* and constructs from "peerDependencies": { ... }, Add { "aws-cdk-lib": "^2.0.0-rc.1", "constructs": "^10.0.0" }
+// 4 - Add { "aws-cdk-lib": "2.0.0-rc.1", "constructs": "^10.0.0" } to "devDependencies"
```
[the remark this answers is not in the document]
Comment (hnishar) [reply deployment/v2/align-version.js:6]: Steps to go back to v1 is
1 - reset v2/align `deployment/v2/align-version.sh revert`
2 - set v1/align `deployment/align-version.sh`

File: deployment/v2/align-version.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      newdependencies[name] = version.replace(findVersion, replaceVersion);
+    }
+  }
+  return newdependencies;
+}
+
+function processPeerDependencies(section, file) {
+  let newdependencies = {};
+  for (const [ name, version ] of Object.entries(section)) {
+    // Remove all entries starting with @aws-cdk/* and constructs
```
[the remark this answers is not in the document]
Comment (hnishar) [reply deployment/v2/align-version.js:48]: Yes, reverting back the files from git is how you can go back to v1 codebase

File: deployment/v2/build-patterns.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+echo "packaging..."
+time lerna run --bail --stream jsii-pacmak || fail
+
+echo "============================================================================================="
+echo "reverting back versions and updates to package.json for CDK v2..."
+/bin/bash $deployment_dir/align-version.sh revert
+
+echo "============================================================================================="
+echo "reverting back TypeScript import statements for CDK v2..."
+git checkout `find . -name *.ts | grep -v node_modules | grep -v **/*.d.ts`
```
[the remark this answers is not in the document]
Comment (hnishar) [reply deployment/v2/build-patterns.sh:60]: Reverting back the *.ts file from Git which will essentially revert back the rewritten import statements

File: deployment/v2/bump.sh [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+echo "Loading ${deployment_dir}/get-version"
+
+# Rename CHANGELOG.md
+echo "Rename CHANGELOG.md to CHANGELOG.md.bak"
+mv CHANGELOG.md CHANGELOG.md.bak
+echo "Rename CHANGELOG.v2.md to CHANGELOG.md"
+mv CHANGELOG.v2.md CHANGELOG.md
+
+# Rename lerna.json
+echo "Rename source/lerna.json to source/lerna.json.bak"
```
[the remark this answers is not in the document]
Comment (hnishar) [reply deployment/v2/bump.sh:29]: I'm not following here, if the concern is to rename the existing changelog.md to changelog.v1.md and lerna.json to lerna.v1.json that's doable, is it more than that?

File: source/tools/aws-cdk-migration/LICENSE (hunk 1 above)
Comment (hnishar) [reply source/tools/aws-cdk-migration/LICENSE:1]: Unfortunately yes, for the same reason as cdk-integ-tools, it needed to be modified to handle special case of rewriting the import statments for `IConstruct`

**Review by biffgaut — DISMISSED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: db56b0288cf0b0cc6b9b2d18c334137590a3a0db
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8fb86179-18ff-4c0f-aedf-b3c335cf898c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 3ef08e7b67f94141257cb7d5b6bd2bf2de41df68
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d6baacbb-4fd0-4d64-840e-f7f5d9f5c1cd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 7844d679943cf5dc0929004721fed4226626a5db
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=acb74f3c-d1d7-4b75-b63c-1dc567b16b01%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**